test/index.ts

```diff
@@ -457,22 +457,23 @@ function testPickProperties() {
 
 function testOptionalKeys() {
   type cases = [
-    // @ts-expect-error converts to Number and gets its optional keys
     Assert<IsExact<OptionalKeys<number>, never>>,
-    // @ts-expect-error converts to String and gets its optional keys
     Assert<IsExact<OptionalKeys<string>, never>>,
-    // wtf?
-    Assert<IsExact<OptionalKeys<boolean>, () => boolean>>,
-    // @ts-expect-error converts to BigInt and gets its optional keys
+    Assert<IsExact<OptionalKeys<boolean>, never>>,
     Assert<IsExact<OptionalKeys<bigint>, never>>,
-    // wtf?
-    Assert<IsExact<OptionalKeys<symbol>, string | ((hint: string) => symbol) | (() => string) | (() => symbol)>>,
+    Assert<IsExact<OptionalKeys<symbol>, never>>,
     Assert<IsExact<OptionalKeys<undefined>, never>>,
     Assert<IsExact<OptionalKeys<null>, never>>,
     Assert<IsExact<OptionalKeys<Function>, never>>,
     Assert<IsExact<OptionalKeys<Date>, never>>,
     Assert<IsExact<OptionalKeys<Error>, "stack">>,
     Assert<IsExact<OptionalKeys<RegExp>, never>>,
+    Assert<IsExact<OptionalKeys<() => void>, never>>,
+    Assert<IsExact<OptionalKeys<any[]>, never>>,
+    Assert<IsExact<OptionalKeys<[1, 2]>, never>>,
+    Assert<IsExact<OptionalKeys<[1, 2?]>, "1">>,
+    Assert<IsExact<OptionalKeys<{ (): void; a?: 1 }>, "a">>,
+    Assert<IsExact<OptionalKeys<{ (): void; a: 1 }>, never>>,
```

Comment:
I believe you're testing the same thing as on lines L478-479?

Reply:
Not exactly, L475-476 tests **functions** with optional properties (they have a call signature `(): void`), whereas L478-479 tests just plain objects.

Reply:
Could you explain why it's  important? If a function returns `never` (an assertion above), there is no difference between object with / without function?

Reply:
For example, if a `Function` check is added that directly returns `never`, **only** the test case with object containing **both** a call signature and an optional property (L475) would fail, ensuring such changes are blocked.

```ts
export type OptionalKeys<Type extends object> = Type extends Function
  ? never
  : keyof {
      [Key in keyof Type as Type extends Required<Pick<Type, Key>> ? never : Key]: any;
    };
```

Reply:
It makes sense to me, thank you for your explanation! Although I believe the risk of regression with functions is low, I'm happy to leave these test cases